backend_new/test/integration/jurisdiction.e2e-spec.ts

```diff
@@ -0,0 +1,185 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { AppModule } from '../../src/app.module';
+import { PrismaService } from '../../src/services/prisma.service';
+import { jurisdictionFactory } from '../../prisma/seed-helpers/jurisdiction-factory';
+import { JurisdictionCreate } from '../../src/dtos/jurisdictions/jurisdiction-create.dto';
+import { JurisdictionUpdate } from '../../src/dtos/jurisdictions/jurisdiction-update.dto';
+import { IdDTO } from 'src/dtos/shared/id.dto';
+import { LanguagesEnum } from '@prisma/client';
+import { randomUUID } from 'crypto';
+
+describe('Jurisdiction Controller Tests', () => {
+  let app: INestApplication;
+  let prisma: PrismaService;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    prisma = moduleFixture.get<PrismaService>(PrismaService);
+    await app.init();
+  });
+
+  it('testing list endpoint', async () => {
+    const jurisdictionA = await prisma.jurisdictions.create({
+      data: jurisdictionFactory(70),
+    });
+    const jurisdictionB = await prisma.jurisdictions.create({
+      data: jurisdictionFactory(80),
+    });
+
+    const res = await request(app.getHttpServer())
+      .get(`/jurisdictions?`)
+      .expect(200);
+
+    expect(res.body.length).toBeGreaterThanOrEqual(2);
+    const jurisdictions = res.body.map((value) => value.name);
+    expect(jurisdictions).toContain(jurisdictionA.name);
+    expect(jurisdictions).toContain(jurisdictionB.name);
+  });
+
+  it("retrieve endpoint with id that doesn't exist should error", async () => {
+    const id = randomUUID();
+    const res = await request(app.getHttpServer())
+      .get(`/jurisdictions/${id}`)
+      .expect(404);
+    expect(res.body.message).toEqual(
+      `jurisdiction ${id} was requested but not found`,
+    );
+  });
+
+  it('testing retrieve endpoint', async () => {
+    const jurisdictionA = await prisma.jurisdictions.create({
+      data: jurisdictionFactory(101),
+    });
+
+    const res = await request(app.getHttpServer())
+      .get(`/jurisdictions/${jurisdictionA.id}`)
+      .expect(200);
+
+    expect(res.body.name).toEqual(jurisdictionA.name);
+  });
+
+  it("retrieve endpoint with id that doesn't exist should error", async () => {
```

Comment:
The test two above has an identical name

Reply:
nice catch! updated!